### PR TITLE
Add frontend numpy require

### PR DESCRIPTION
**Pull request by ahmedelgamal0**

### CODE REVIEW

**Comment by Dsantra92** [comment]
Hi @ahmedelgamal0 thanks for your contribution 😄 . If you are making a pr for closing a issue mention `closes #<issue-number>`. See [ivy docs](https://unify.ai/docs/ivy/overview/contributing/the_basics.html#todo-list-issues) for more details.

**Comment by ahmedelgamal0** [comment]
Close #16463

**Comment by Dsantra92** [comment]
Hi @ahmedelgamal0 it should be `Close #<issue-number>` not the pr number. You need to include the issue number you are trying to solve. You can refer to [github issue tutorial](https://www.youtube.com/watch?v=TKJ4RdhyB5Y) for finding the same.

File: ivy_tests/test_ivy/test_frontends/test_numpy/test_manipulation_routines/test_changing_array_shape.py [hunk 1]
```diff
@@ -282,3 +282,33 @@ def test_numpy_asarray_chkfinite(
...
     )
+
+#require
+@handle_frontend_test(
+    fn_tree="numpy.require",
+    dtype_and_a=helpers.dtype_and_values(available_dtypes=helpers.get_dtypes("float")),
+    dtype_and_requirements=helpers.dtype_and_values(available_dtypes=helpers.get_dtypes("float")),
+    test_with_out=st.just(False),
+)
+def test_numpy_require(
```
Comment (Sam-Armstrong) [inline ivy_tests/test_ivy/test_frontends/test_numpy/test_manipulation_routines/test_changing_array_shape.py:293]: It seems that the test you wrote is failing, please can you fix this? 

The test can be run with the command `pytest ivy_tests/test_ivy/test_frontends/test_numpy/test_manipulation_routines/test_changing_array_shape.py::test_numpy_require`

**Review by Sam-Armstrong — CHANGES_REQUESTED** [review]
Hi @ahmedelgamal0, thanks for making the PR! I've left some comments, please could you address these when you get chance? 🙂

**Comment by Sam-Armstrong** [comment]
@ahmedelgamal0 I've fixed the test and formatting so we can get this merged. Thanks for your work for this PR! 😊

**Review by Sam-Armstrong — APPROVED** [review]

**Pull request merged**